src/server.cpp

```diff
@@ -2760,15 +2682,10 @@ void Server::sendDetachedInventory(const std::string &name, session_t peer_id)
 
 void Server::sendDetachedInventories(session_t peer_id, bool incremental)
 {
-	for (const auto &detached_inventory : m_detached_inventories) {
-		const std::string &name = detached_inventory.first;
-		if (incremental) {
-			Inventory *inv = detached_inventory.second;
-			if (!inv || !inv->checkModified())
-				continue;
-		}
-
-		sendDetachedInventory(name, peer_id);
+	std::unordered_map<std::string, Inventory *> inventories;
+	m_inventory_mgr->getDetachedInventories(inventories, incremental);
```

Comment:
Rather than copying the map, I'd prefer to use a lambda expression as callback here.

Reply:
fine, i will do that tomorrow